torch/csrc/autograd/profiler.cpp

```diff
@@ -468,21 +467,29 @@ void enableProfiler(const ProfilerConfig& new_config) {
   state->mark("__start_profile", false);
 }
 
-thread_event_lists disableProfiler() {
+thread_event_lists disableProfiler(bool cleanupTLSState, bool consolidate) {
   // all the DebugInfoBase objects are scope based and supposed to use DebugInfoGuard
-  auto state = c10::ThreadLocalDebugInfo::_pop(c10::DebugInfoKind::PROFILER_STATE);
+  std::shared_ptr<c10::DebugInfoBase> state;
+  if (cleanupTLSState) {
+    state = c10::ThreadLocalDebugInfo::_pop(c10::DebugInfoKind::PROFILER_STATE);
```

Comment:
nit question: do we have to pop here? Will it work if this line always uses peek, and let the `if (cleanupTLSState)` decide whether it should pop, to save one branch?

Reply:
In the majority use case where `cleanupTLSState = true`, that would incur both a pop and a peek, whereas the current would have an extra branch but only do one of pop/peek, right?